model/task_lifecycle.go

```diff
@@ -559,8 +557,7 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 		}
 	}
 
-	// if there are no failed tasks, mark the build as started
-	if !failedTask {
+	if b.Status == evergreen.BuildCreated {
```

Comment:
Suppose the build has the status `evergreen.BuildCreated` from 496. A task fails. On 535 we update the status to `evergreen.BuildFailed` in the database. Then we get here. We will change the status to `evergreen.BuildStarted` in the db, incorrectly. I think 535 also needs to update the in-memory build to avoid that case.

Reply:
build.UpdateStatus updates the in-memory build. Is this still an issue?